### PR TITLE
Code fixes2

### PR DESCRIPTION
**Pull request by Avi-Patel**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,2 +1,3 @@
 {
+  "liveServer.settings.port": 5501
```
Comment (jaikiratsingh) [inline .vscode/settings.json:2]: This file should be in `.gitignore`

File: src/js/consts.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+export const todoActions = {
```
Comment (jaikiratsingh) [inline src/js/consts.js:1]: rename file to `constants.js`

File: src/js/controller.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      currentlySelectedIds,
+      this.callbacksForTodo
+    );
+    filteredTodos.forEach((todo) =>
+      this.view.analyticsUpdater.updateCounts(todo)
+    );
+    this.view.analyticsUpdater.updateAnalyticsOnView();
+  };
+
+  createTodoObject = ({ counter, title, urgency, category }) => {
```
Comment (jaikiratsingh) [inline src/js/controller.js:54]: defaults should be handled as
```js
createTodoObject = ({counter, title='', urgency=0, category=0}) => {
```

Also, use `URGENCY.LOW` / `CATEGORY.LOW` as the default and not 0

File: src/js/controller.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+  handleFilterUpdate = (urgencyOrCategory, level) => {
+    let index = 0;
+    if (level === urgency.MEDIUM || level === category.ACADEMIC) {
+      index = 1;
+    }
+    if (level == urgency.HIGH || level === category.SOCIAL) {
+      index = 2;
+    }
+    this.model.updateFilter(urgencyOrCategory, index);
```
Comment (jaikiratsingh) [inline src/js/controller.js:90]: what is `index` for ? index=1 and index=2 doesn't tell me what their purpose is.

File: src/js/controller.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    this.model.updateCheckBoxStatus(isChecked);
+  };
+
+  handleSearchBoxUpdate = (searchInput) => {
+    this.model.updateSearchedInput(searchInput);
+  };
+
+  handleEditTodo = (todo) => {
+    const index = this.model.findIndexById(todo.ID);
+    this.model.replaceTodoAtAnyIndex(index, todo);
```
Comment (jaikiratsingh) [inline src/js/controller.js:103]: `replaceTodoAtIndex`

File: src/js/controller.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  };
+
+  handleTodoToggle = (id) => {
+    const index = this.model.findIndexById(id);
+    const todoAtIndex = this.model.getTodo(index);
+    const updatedTodo = { ...todoAtIndex, completed: !todoAtIndex.completed };
+    this.model.replaceTodoAtAnyIndex(index, updatedTodo);
+  };
+
+  handleTogglingFromSelected = (id) => {
```
Comment (jaikiratsingh) [inline src/js/controller.js:113]: `handleSelectionToggle` or something of that sort

File: src/js/controller.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    this.model.toggleIdFromSelected(id);
+  };
+
+  handleDeleteTodo = (id) => {
+    this.model.deleteTodo(id);
+  };
+
+  toggleCompletionOfSelection = () => {
```
Comment (jaikiratsingh) [inline src/js/controller.js:121]: `toggleBulkCompletion`

File: src/js/controller.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+    } else {
+      this.model.deleteTodoInBulk();
+    }
+  };
+
+  handleUndo = () => {
+    const position = this.model.getPosition();
+    const actions = this.model.getActions();
+    console.log(actions);
+    if (position === -1) {
```
Comment (jaikiratsingh) [inline src/js/controller.js:145]: Replace `-1` with a sentinal. If it is an invalid position, create a constant named `INVALID_POSITION`

File: src/js/controller.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+        this.model.onEdit(actions[position], true);
+        break;
+      case todoActions.CREATE:
+        this.model.onCreate(actions[position], true);
+        break;
+      case todoActions.DELETE:
+        this.model.onDelete(actions[position], true);
+        break;
+      default:
+        break;
```
Comment (jaikiratsingh) [inline src/js/controller.js:161]: default statement doesn't need a `break` statement

File: src/js/filterValidationOnTodo.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+export const validateTodo = (todo, filterData) => {
```
Comment (jaikiratsingh) [inline src/js/filterValidationOnTodo.js:1]: `validateTodoForFilter` might be a better name

File: src/js/model.js [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+    return index;
+  };
+
+  removeCurrentlySelectedIds = () => {
+    this.currentlySelected = [];
+    this.runStateChangeHandler();
+  };
+
+  setCurrentlySelected = (newSelectedIds) =>
+    (this.currentlySelected = [...newSelectedIds]); //
```
Comment (jaikiratsingh) [inline src/js/model.js:56]: Why spread operator ?

File: src/js/model.js [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+
+  getCurrentlySelected = () => this.currentlySelected;
+
+  insertTodoAtAnyIndex = (index, todo) => this.allTodos.splice(index, 0, todo);
```
Comment (jaikiratsingh) [inline src/js/model.js:60]: `insertTodoAtIndex`

File: src/js/model.js [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+    if (indexInSelected === -1) {
+      this.currentlySelected = this.currentlySelected.concat(id);
+    } else {
+      this.currentlySelected.splice(indexInSelected, 1);
+    }
+    this.runStateChangeHandler();
+  };
+
+  updateFilter = (urgencyOrCategory, index) => {
+    if (urgencyOrCategory === "urgency") {
```
Comment (jaikiratsingh) [inline src/js/model.js:106]: use constant

File: src/js/model.js [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+      .then(() => {
+        this.addActions(
+          todoActions.EDIT,
+          [todo.ID],
+          [todo],
+          [this.allTodos[index]]
+        );
+        this.allTodos = this.allTodos
+          .slice(0, index)
+          .concat({ ...todo }, this.allTodos.slice(index + 1));
```
Comment (jaikiratsingh) [inline src/js/model.js:171]: why spread ?

File: src/js/views/createModal.js [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+    if (updatedTitle.trim() !== "") {
+      const updatedTodo = { ...todo };
+      updatedTodo.title = updatedTitle;
+      updatedTodo.urgency = updateModal.querySelector(
+        "#updated-urgency"
+      ).selectedIndex;
+      updatedTodo.category = updateModal.querySelector(
+        "#updated-category"
+      ).selectedIndex;
```
Comment (jaikiratsingh) [inline src/js/views/createModal.js:14]:
```js
const updatedUrgency = updateModal.querySelector("#updated-urgency").selectedIndex;
const updatedCategory = updateModal.querySelector("#updated-category").selectedIndex;

const updatedTodo = {...todo, title: updatedTitle, category: updatedCategory, urgency: updatedUrgency};
```

File: src/js/views/createModal.js [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+  });
+};
+
+const addListenerToModalCancelBtn = (btnID, updateModal) => {
+  const cancelBtn = document.querySelector(`#${btnID}`);
+  cancelBtn.addEventListener("click", () => updateModal.remove());
+};
+
+const createModal = (title, urgencyIndex, categoryIndex) => {
+  const urgency = ["low", "medium", "high"];
```
Comment (jaikiratsingh) [inline src/js/views/createModal.js:28]: use constants

File: src/js/views/createModal.js [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+  const category = ["personal", "academic", "social"];
```
Comment (jaikiratsingh) [inline src/js/views/createModal.js:29]: use constants

**Review by jaikiratsingh — COMMENTED** [review]
Please see the following suggested changes

File: src/js/consts.js (hunk 2 above)
Comment (Avi-Patel) [reply src/js/consts.js:1]: done

File: src/js/controller.js (hunk 3 above)
Comment (Avi-Patel) [reply src/js/controller.js:54]: handled defaults in the function called "handleAddTodo" where i am getting inputs from the view.

**Pull request merged**